lib/ansible/modules/remote_management/redfish/redfish_facts.py

```diff
@@ -0,0 +1,251 @@
+    get information back.
+  - Information retrieved is placed in a location specified by the user.
+options:
+  category:
+    required: true
+    description:
+      - Action category to execute on server
+    choices: ["System", "Chassis", "Accounts", "Update", "Manager"]
+  command:
+    required: true
+    description:
+      - Command to execute on server
+  baseuri:
+    required: true
+    description:
+      - Base URI of OOB controller
+  user:
+    required: true
+    description:
+      - User for authentication with OOB controller
+  password:
+    required: true
+    description:
+      - Password for authentication with OOB controller
+
+author: "Jose Delarosa (github: jose-delarosa)"
+'''
+
+EXAMPLES = '''
+  - name: Get system inventory
+    redfish_facts:
+      category: System
+      command: GetCpuInventory
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Get system inventory (default command)
+    redfish_facts:
+      category: System
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Get system inventory (more than one command)
+    redfish_facts:
+      category: System
+      command: "GetNicInventory GetPsuInventory GetBiosAttributes"
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Get fans statistics
+    redfish_facts:
+      category: Chassis
+      command: GetFanInventory
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: List all users
+    redfish_facts:
+      category: Accounts
+      command: ListUsers
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+'''
+
+RETURN = '''
+result:
+    description: different results depending on task
+    returned: always
+    type: dict
+    sample: List of CPUs on system
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.redfish_utils import RedfishUtils
+
+CATEGORY_COMMANDS_ALL = {
+    "System": ["GetSystemInventory", "GetPsuInventory", "GetCpuInventory",
+               "GetNicInventory", "GetStorageControllerInventory",
+               "GetDiskInventory", "GetBiosAttributes", "GetBiosBootOrder"],
+    "Chassis": ["GetFanInventory"],
+    "Accounts": ["ListUsers"],
+    "Update": ["GetFirmwareInventory"],
+    "Manager": ["GetManagerAttributes", "GetLogs"],
+}
+
+CATEGORY_COMMANDS_DEFAULT = {
+    "System": "GetSystemInventory",
+    "Chassis": "GetFanInventory",
+    "Accounts": "ListUsers",
+    "Update": "GetFirmwareInventory",
+    "Manager": "GetManagerAttributes",
+}
+
+
+def main():
+    result_all = {}
+    result = {}
+    cmd_list = []
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, choices=["System", "Chassis", "Accounts", "Update", "Manager"]),
```

Comment:
Rather than make this required, set a default value. This will allow the facts module to work with minimal configuration.

Reply:
I believe this should be either `list` or `string` type, not `dict`.

Would it be possible to make this field accept a list of categories to gather? The `if..elif` logic later on wolud need to change, but this would allow all the data to be gathered by one task rather than having to run the fact task multiple times.